lib/spack/spack/test/views.py

```diff
@@ -23,3 +24,55 @@ def test_remove_extensions_ordered(install_mockery, mock_fetch, tmpdir):
 
     e1 = e2["extension1"]
     view.remove_specs(e1, e2)
+
+
+@pytest.mark.regression("32456")
+def test_view_with_spec_not_contributing_files(install_mockery, mock_fetch, tmpdir):
+    # This test is a bit contrived, but it checks whether
+    # pkg.add_files_to_view() receives the correct source paths
+    # when another package does not have any files to
+    # contribute to a view. That can happen either because
+    # there simply are no files in the prefix, or because all
+    # its files conflict with another spec -- here we test the
+    # former).
+
+    view_dir = str(tmpdir)
+    layout = DirectoryLayout(view_dir)
+    view = SimpleFilesystemView(view_dir, layout)
+
+    a = Spec("libelf").concretized()
+    b = Spec("trivial-install-test-package").concretized()
+    a.package.do_install()
+    b.package.do_install()
+
+    # Remove all files of pkg a
+    for root, _, files in os.walk(a.prefix):
+        for name in files:
+            os.remove(os.path.join(root, name))
+
+    # Create some dirs
+    os.makedirs(a.prefix.first)
+    os.makedirs(a.prefix.second)
+    os.makedirs(b.prefix.second)
+    os.makedirs(b.prefix.third)
+
+    def pkg_a_add_files_to_view(view, merge_map, skip_if_exists=True):
+        assert False, "There shouldn't be files to add"
+
+    def pkg_b_add_files_to_view(view, merge_map, skip_if_exists=True):
+        pkg_b_add_files_to_view.is_called = True  # ugh
+
+        # There should be a single file.
+        assert len(merge_map) == 1
+        src, dst = list(merge_map.items())[0]
+        assert os.path.basename(src) == os.path.basename(dst) == "an_installation_file"
+        assert os.path.samefile(os.path.dirname(src), b.prefix)
+        assert os.path.samefile(os.path.dirname(dst), view_dir)
+
+    a.package.add_files_to_view = pkg_a_add_files_to_view
+    b.package.add_files_to_view = pkg_b_add_files_to_view
+
+    # Check if the dirs end up in the view.
+    view.add_specs(a, b)
+
+    assert pkg_b_add_files_to_view.is_called
```

Comment:
Better to inspect the view I think: make sure the file is there

Reply:
The idea was to unit test `add_files_to_view`, but the suggestion is fine to me.